### DISCUSSION TITLE
I wrote a plugin to rewrite a value, to be a link to a Foreign Key, and outlink or whatever.  I have no idea if it can be accepted...

### DISCUSSION THREAD

**Discussion by RoyBellingan** [adminerevo/adminerevo#89] [Show and tell]
Is done mostly via JS as they are about interactive functionality on the front end

The basic configuration is like

```JSON
{
  //Source DB
  "test": {
    //Source Table
    "a": {
      //Source Column
      "v": {
              //several parameter that will toggle the behaviour
      }
    }
  }
}
```

* abilty to have "custom" event for certain column, for example I use it to create on the fly link based on the column value, this is done defining a custom rewrite function 

* allowing the "click to foreign key" abilty also for view mostly.
Of course it also allow linking other column as you wish...
the block will be something like, 
you can use a custom function, as sometimes I want to link to multiple table
```JSON
"v": {
        //Target Table
        "table": "amarokdb.tracks",
        //Column that will be used for the target
        "id": "title"
}
```

If you pass some additional parameter, it will became a select2 box, and allow to FILL based on that column value
```JSON

      "v": {
        //Target Table
        "table": "amarokdb.tracks",
        //Column that will be used for the select VALUE field
        "id": "title",
        "smartKey" : "remoteColumn"
        "text" : "some name or suggestion" 
        "where": "AND destinationId IS NOT NULL" //can be optionally prefiltered
      }

```

What do you think ? If you think this can be accepted I will try to clean up the code, which at the moment is orrible and full of hardwired stuff.

**Comment by LionelLaffineur** [comment]
Hi @RoyBellingan,
Idea seems promising.  Would you paste some screenshots here so we can have a better idea of the outcome ?
Regards,
L.

**Comment by RoyBellingan** [comment]
So this config json
```JSON
{
    "SCCE_Common": {
        "sourceAccount" : {
            "ownerId": {
                "table" : "SCCE_Common.owner",
                "id" : "id",
                "text" : "name"
            }
        }
    }
}

```
Reads from this table

![image](https://github.com/adminerevo/adminerevo/assets/368572/fecd94cc-3194-44db-9911-4a67b5cb0f44)

And produces that when updating the sourceAccount table

![image](https://github.com/adminerevo/adminerevo/assets/368572/c84597df-4930-4669-9e8f-70d50f5ca3b8)

The dropdown is a https://select2.org one and is already wired to provide filtering (even with a custom function) 

![image](https://github.com/adminerevo/adminerevo/assets/368572/82b6832c-f5bd-4a98-bcbb-b7b16f9454f6)


********

For the "hotlinking" part, well... the column will just have a highlight or a custom content so something like (this is another system atm I will merge soon)

```JSON

  "turboProp":{
	"urlDecoder":{
		"destName":[
			"imageLinkerDirect1"
		],
		"hostPattern":[
		         "imageLinker1"
		]
	}
  }

```


![image](https://github.com/adminerevo/adminerevo/assets/368572/2849db58-4001-4f2e-89df-e520bacf1f48)

**Comment by LionelLaffineur** [comment]
For the first part, I follow.  The config is quite clear.
For the second part, can we see it later ?
If you can do a PR for the FK selector, I will test it and accept when it's OK.
We'll discuss the second when finished.

Thanks already for your interest in the tool and your contribution !

**Comment by RoyBellingan** [comment]
So for this initial tech demo ok, the problem is that is mostly Javascript at the moment. So if layout changes will not work. Also for the moment will work using a small php module, which is just for convenience, if there is interest I will integrate better.
I am looking if I can integrate mode stuff in the php side.

**Comment by LionelLaffineur** [comment]
Perhaps can you already propose a draft PR ?

**Comment by RoyBellingan** [comment]
Ciao Lionel, so finally had some time (was with Covid home) !

Here is the current version 
https://github.com/RoyBellingan/adminerevo/blob/cbh/plugins/cbh/js/cbh.js

I tried to find a way to have an ajax "interface", but I was not able to load the internal machiner from "a file that is not admin.php", because all the include are relative, so I had to, well do that

https://github.com/RoyBellingan/adminerevo/blob/cbh/adminer/download.inc.php

Basically took something close enought to what I needed, and I just extended, I would have prefered to have a file inside my folder act as the ajax backend, but I do not think is possible.

I wish soon to add a public demo to show case it.
